agent/reconciler/host_reconciler.go

```diff
@@ -94,12 +98,38 @@ func (r *HostReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctr
 
 func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
 	logger := ctrl.LoggerFrom(ctx)
+	logger.Info("reconcile normal")
 	if byoHost.Status.MachineRef == nil {
 		logger.Info("Machine ref not yet set")
 		conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.WaitingForMachineRefReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
 	}
 
+	if r.UseInstallerController && !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded) {
+		if byoHost.Spec.InstallationSecret == nil {
+			logger.Info("InstallationSecret not ready")
+			conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded, infrastructurev1beta1.K8sInstallationSecretUnavailableReason, clusterv1.ConditionSeverityInfo, "")
+			return ctrl.Result{}, nil
+		}
+		installScript, uninstallScript, err := r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret.Name, byoHost.Spec.InstallationSecret.Namespace)
+		if err != nil {
+			logger.Error(err, "error getting installation script")
+			return ctrl.Result{}, err
+		}
+		byoHost.Spec.UninstallationScript = &uninstallScript
+		installScript, err = r.parseInstallationScript(ctx, installScript)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		logger.Info("executing install script")
+		err = r.executeScript(ctx, installScript)
+		if err != nil {
+			logger.Error(err, "error execting installation script")
+			return ctrl.Result{}, err
+		}
```

Comment:
if [installK8sComponents()](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/reconciler/host_reconciler.go#L126) is failing then we are doing following
```sh
r.Recorder.Event(byoHost, corev1.EventTypeWarning, "InstallK8sComponentFailed", "k8s component installation failed")
conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded, infrastructurev1beta1.K8sComponentsInstallationFailedReason, clusterv1.ConditionSeverityInfo, "")			
```
Can we do same thing in this new flow as well?

Reply:
Do I need to change the test cases as well after adding this?